src/components/application_manager/src/commands/hmi/request_to_hmi.cc

```diff
@@ -36,6 +36,29 @@ namespace application_manager {
 
 namespace commands {
 
+bool CheckAvailabilityHMIInterfaces(ApplicationManager& application_manager,
+                                    HmiInterfaces::InterfaceID interface) {
+  const HmiInterfaces::InterfaceState state =
+      application_manager.hmi_interfaces().GetInterfaceState(interface);
+  return HmiInterfaces::STATE_NOT_AVAILABLE != state ? true : false;
+}
+
+bool ChangeInterfaceState(ApplicationManager& application_manager,
+                          const smart_objects::SmartObject& response_from_hmi,
+                          HmiInterfaces::InterfaceID interface) {
+ bool is_available = false;
+ if (response_from_hmi[strings::msg_params].keyExists(strings::available)) {
+   is_available = response_from_hmi[strings::msg_params][strings::available].asBool();
+   const HmiInterfaces::InterfaceState interface_state =
+       is_available ? HmiInterfaces::STATE_AVAILABLE
+                    : HmiInterfaces::STATE_NOT_AVAILABLE;
+   application_manager.hmi_interfaces().SetInterfaceState(
+         interface, interface_state);
+ }
+ return is_available;
```

Comment:
@pestOO I used this variable in order to write value from smart object to it, and after that I used this value in order to determine state for interfaces.